app/controllers/simplified-account/settings/card-types/card-types.controller.js

```diff
@@ -1,22 +1,71 @@
 const { response } = require('@utils/response')
 const { formatCardTypesForTemplate } = require('@utils/simplified-account/format/format-card-types')
-const { getAllCardTypes, getAcceptedCardTypesForServiceAndAccountType } = require('@services/card-types.service')
+const { getAllCardTypes, getAcceptedCardTypesForServiceAndAccountType, postAcceptedCardsForServiceAndAccountType } = require('@services/card-types.service')
+const { formatSimplifiedAccountPathsFor } = require('@utils/simplified-account/format')
+const paths = require('@root/paths')
 
 async function get (req, res, next) {
-  const serviceId = req.service.externalId
+  const serviceExternalId = req.service.externalId
   const accountType = req.account.type
-  const isAdminUser = req.user.isAdminUserForService(serviceId)
+  const isAdminUser = req.user.isAdminUserForService(serviceExternalId)
+  const messages = res.locals?.flash?.messages ?? []
   try {
     const { card_types: allCards } = await getAllCardTypes()
-    const { card_types: acceptedCards } = await getAcceptedCardTypesForServiceAndAccountType(serviceId, accountType)
+    const { card_types: acceptedCards } = await getAcceptedCardTypesForServiceAndAccountType(serviceExternalId, accountType)
+    const currentAcceptedCardTypeIds = acceptedCards.map(card => card.id)
     const cardTypes = formatCardTypesForTemplate(allCards, acceptedCards, req.account, isAdminUser)
-    response(req, res, 'simplified-account/settings/card-types/index',
-      { cardTypes, isAdminUser })
+    response(req, res, 'simplified-account/settings/card-types/index', {
+      messages,
+      cardTypes,
+      isAdminUser,
+      currentAcceptedCardTypeIds
+    })
+  } catch (err) {
+    next(err)
+  }
+}
+
+async function post (req, res, next) {
+  const serviceExternalId = req.service.externalId
+  const accountType = req.account.type
+  const selectedDebitCards = (typeof req.body.debit === 'string' ? [req.body.debit] : req.body.debit) || []
+  const selectedCreditCards = (typeof req.body.credit === 'string' ? [req.body.credit] : req.body.credit) || []
```

Comment:
we can replace these with express validation chains like so:
```js
body('credit')
  .notEmpty()
  .customSanitizer(value => Array.isArray(value) ? value : [value])
  .isArray({ min: 1 })
  .withMessage('You must choose at least one card')
  .customSanitizer(value => {
    req.selectedCreditCards = value
    return value
  })
  ```
this might be slightly more LOC (you'll need a debit card one also) but matches our other controllers and in my opinion is easier to understand

Reply:
done - yes this does add a bit more code but maybe maintaining consistency is worth it.

Reply:
sanitiseArray is now only used once for debit cards and once for credit cards.